### PR TITLE
Feat archivo

### PR DESCRIPTION
**Pull request by mixcas**
- add archivo scaffolding
- add filter by year and post type / exhibition type

I'm not sure about the way resetting the filter works in the types because that functionality is not available to year filter. A different approach is adding an `All` link.

Shit, I just noticed i added `footer.php` to this commits :(
I could remove it, but i did that cuz when the internet sux at the studio the page gets stuck loading jquery from CDN…

### CODE REVIEW

File: css/site.styl [hunk 1]
```diff
@@ -218,6 +218,22 @@ img
 
 // FOOTER
 
+// ARCHIVO
+// YEARS and FILTERS
+#year-filter a,
```
Comment (varffs) [inline css/site.styl]: this prob need to be replicable code as used in other archives

File: functions.php [hunk 2]
```diff
@@ -125,6 +125,103 @@ function custom_login_logo() {
...
+
+  $terms = array();
+
+  foreach($posts->posts as $post) {
+
+    if ($post->post_type === 'exposiciones') {
+      $post_type = get_post_type_object( $post->post_type );
+      $post_terms = wp_get_post_terms( $post->ID, 'tipo_de_exposicion' );
+      foreach($post_terms as $post_term) {
+        echo in_array($post_term->slug, array_column($terms, 'slug'));
```
Comment (varffs) [inline functions.php]: should this echo still be here?

File: functions.php [hunk 3]
```diff
@@ -125,6 +125,103 @@ function custom_login_logo() {
...
+        $terms[] = array(
+          'slug'  =>  $post_type->name,
+          'name'  =>  $post_type->label,
+        );
+      }
+    }
+
+  }
+  //pr( $terms );
+  return $terms;
```
Comment (varffs) [inline functions.php:166]: this function looks very similar to the previous. Perhaps better to make 1 function with another variable?

File: functions.php [hunk 4]
```diff
@@ -125,6 +125,103 @@ function custom_login_logo() {
...
+  $where .= ' ) AND post_status = "publish " ';
+
+  $order = strtoupper( $order );
+  if ( $order !== 'ASC' ) {
+    $order = 'DESC';
+  }
+
+  $years = array();
+
+  $last_changed = wp_cache_get( 'last_changed', 'posts' );
```
Comment (varffs) [inline functions.php:197]: can you explain this block in comments please

File: functions.php [hunk 5]
```diff
@@ -125,6 +125,103 @@ function custom_login_logo() {
...
+  $key = "igv_get_all_years:$key:$last_changed";
+  if ( ! $results = wp_cache_get( $key, 'posts' ) ) {
+    $results = $wpdb->get_results( $query );
+    wp_cache_set( $key, $results, 'posts' );
+  }
+  if ( $results ) {
+    foreach ( (array) $results as $result) {
+      $years[] = $result->year;
+    }
+  }
```
Comment (varffs) [inline functions.php:221]: no handling for no results?

File: lib/post-types.php [hunk 6]
```diff
@@ -65,6 +65,13 @@ function post_type_exposiciones() {
 	);
 	register_post_type( 'exposiciones', $args );
 
+  add_rewrite_rule(
+    'exposiciones/([0-9]{4})/?$',
+    'index.php?post_type=obra&year=$matches[1]',
```
Comment (varffs) [inline lib/post-types.php]: post_type obra?

File: page-archivo.php [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+$archivo_post_types = array(
+  'post',
+  'exposiciones',
+  'obra',
+  'journal',
+  'publicaciones',
+);
+
+// Get yearl url param
+$year_param = empty($_GET['año']) ? null : $_GET['año'];
```
Comment (varffs) [inline page-archivo.php]: I cannot believe we are using an `ñ`

File: functions.php (hunk 2 above)
Comment (mixcas) [inline functions.php]: oh shit no

File: page-archivo.php (hunk 7 above)
Comment (varffs) [inline page-archivo.php]: rly think better to just use an `a` and not take any risks

File: page-archivo.php [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+?>
+        <li><a href="#" data-filter="<?php echo $filter_term['slug']; ?>"><?php echo $filter_term['name']; ?></a></li>
+<?php
+}
+?>
+      </ul>
+    </div>
+
+    <!-- main posts loop -->
+    <section id="posts" class="col col-14 u-float">
```
Comment (varffs) [inline page-archivo.php]: doesnt need `u-float`

File: page-archivo.php [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+<?php
+  }
+} else {
+?>
+    <article class="u-alert"><?php _e('Sorry, no posts matched your criteria :{'); ?></article>
+<?php
+} ?>
+
+    <?php get_template_part('partials/pagination'); ?>
```
Comment (varffs) [inline page-archivo.php]: does this page paginate?

File: page-archivo.php (hunk 7 above)
Comment (mixcas) [inline page-archivo.php]: ashhh ¬¬

Comment (mixcas) [inline page-archivo.php]: party pooper

File: lib/post-types.php (hunk 6 above)
Comment (varffs) [inline lib/post-types.php]: should it not be exposiciones?

File: page-archivo.php (hunk 7 above)
Comment (varffs) [inline page-archivo.php]: all fun and games until someones shit browser doesnt work and then the client complains

File: functions.php (hunk 5 above)
Comment (mixcas) [inline functions.php:221]: its fine like that. i got the pattern from the wp core

File: functions.php [hunk 10]
```diff
@@ -125,6 +125,99 @@ function custom_login_logo() {
...
+  $order = strtoupper( $order );
+  if ( $order !== 'ASC' ) {
+    $order = 'DESC';
+  }
+
+  $years = array();
+
+  // The 'last_changed' incrementor is used to invalidate the $key cache. 
+  // This way we invalidate the cache on add, delete, and update.
+  $last_changed = wp_cache_get( 'last_changed', 'posts' );
```
Comment (varffs) [inline functions.php:197]: ah ok i get it now. and this cache value is just for any change event on any post

File: functions.php [hunk 11]
```diff
@@ -125,6 +125,99 @@ function custom_login_logo() {
...
+  if ( ! $last_changed ) {
+    $last_changed = microtime();
+    wp_cache_set( 'last_changed', $last_changed, 'posts' );
```
Comment (varffs) [inline functions.php:200]: but why do you need to hard set here? or is this just for when the cache has expired or something?

File: functions.php [hunk 12]
```diff
@@ -125,6 +125,99 @@ function custom_login_logo() {
...
+  }
+
+  // Final query
+  $query = "SELECT YEAR(post_date) AS `year` FROM $wpdb->posts $where GROUP BY YEAR(post_date) ORDER BY post_date $order";
+
+  // By creating a cache key that's a hash of igv + last_changed + md5 (from query)
+  // we have a simple method for cache invalidation: whenever new activity 
+  // (or whatever) is created, bump last_changed. Now all cache using a key 
+  // generated from last_changed is invalidated.
```
Comment (varffs) [inline functions.php:209]: nice

**Comment by varffs** [comment]
There is no `all` filter?

**Comment by varffs** [comment]
Should this PR include all the markup & styling or just the functionality? I don't really mind either way but not sure what others think?

**Comment by varffs** [comment]
Functionality is all there. Still using `año` though which when copied as a url from Chrome escapes the ñ and looks a bit funky `http://localhost:8888/wp/gonzalo/en/archivo/?a%C3%B1o=2001`. Works fine in all Mac browsers though. Feel worth testing in IE.

File: functions.php (hunk 11 above)
Comment (mixcas) [inline functions.php:200]: cuz this move count as a change

**Comment by mixcas** [comment]
`all` as for all posts? cuz that includes pages and shit, that why i didnt use it

i tought año was gone wtf. ill check

**Comment by mixcas** [comment]
@varffs nah bro, `año` is gone
also my chrome doesn't escape it :)

**Comment by mixcas** [comment]
i think style in diff branch

**Comment by varffs** [comment]
all as in the all filter in the design. same as the initial state for a year archive

**Comment by varffs** [comment]
chrome only escapes when u copy the link

**Comment by mixcas** [comment]
ohhhh its at the bottom, i never saw it. 

im on it

**Comment by mixcas** [comment]
i forgot localization, hold on

**Pull request merged**